tests/test_interface_private_vlan.rb

```diff
@@ -0,0 +1,836 @@
+          assert_equal(start, interface.switchport_mode_private_vlan_trunk,
+                       "Err: Switchport mode, #{start}, not as expected")
+        rescue Cisco::CliError
+          next
+        end
+      end
+    end
+  end
+
+  def test_interface_no_switchport_private_host_mode
+    if validate_property_excluded?('interface',
+                                   'switchport_mode_private_vlan_host')
+      assert_nil(interface.switchport_mode_private_vlan_host)
+      assert_raises(Cisco::UnsupportedError) do
+        interface.switchport_mode_private_vlan_host = :host
+      end
+      return
+    else
+      switchport_modes = [
+        :host,
+        :promiscuous,
+      ]
+
+      switchport_modes.each do |start|
+        begin
+          interface.switchport_mode_private_vlan_host = start
+          assert_equal(start, interface.switchport_mode_private_vlan_host,
+                       "Err: Switchport mode, #{start}, not as expected")
+          interface.switchport_mode_private_vlan_host = :disabled
+          assert_equal(:disabled, interface.switchport_mode_private_vlan_host,
+                       'Error: Switchport mode not disabled')
+        rescue Cisco::CliError
+          next
+        end
+      end
+    end
+  end
+
+  def test_interface_no_switchport_private_trunk_mode
+    if validate_property_excluded?('interface',
+                                   'switchport_mode_private_vlan_trunk')
+      assert_nil(interface.switchport_mode_private_vlan_trunk)
+      assert_raises(Cisco::UnsupportedError) do
+        interface.switchport_mode_private_vlan_trunk = :secondary
+      end
+      return
+    else
+      switchport_modes = [
+        :promiscuous,
+        :secondary,
+      ]
+
+      switchport_modes.each do |start|
+        begin
+           interface.switchport_mode_private_vlan_trunk = start
+           assert_equal(start, interface.switchport_mode_private_vlan_trunk,
+                        "Err: Switchport mode, #{start}, not as expected")
+           interface.switchport_mode_private_vlan_trunk = :disabled
+           assert_equal(:disabled, interface.switchport_mode_private_vlan_trunk,
+                        'Err: Switchport mode not disabled')
+
+         rescue Cisco::CliError
+           next
+         end
+      end
+    end
+  end
+
+  def test_interface_switchport_private_host_association
+    if validate_property_excluded?('interface',
+                                   'switchport_mode_private_vlan_host')
+      assert_nil(interface.switchport_mode_private_vlan_host)
+      assert_raises(Cisco::UnsupportedError) do
+        interface.switchport_mode_private_vlan_host = :host
+      end
+      return
+    else
+      v1 = Vlan.new(10)
+      pv_type = 'primary'
+      v1.private_vlan_type = pv_type
+      v2 = Vlan.new(11)
+      pv_type = 'community'
+      v2.private_vlan_type = pv_type
+      v1.private_vlan_association = ['11']
+      input = %w(10 11)
+      result = %w(10 11)
```

Comment:
nit: you don't need separate variables for `input` and `result` since your input value is your expected value. Just use one, e.g.  `assoc = %w(10 11)`